util/src/main/scala/scala/scalanative/util/NirPrelude.scala

```diff
@@ -0,0 +1,39 @@
+package scala.scalanative
+package util
+
+import java.nio.ByteBuffer
+
+case class NirPrelude(magic: Int,
+                      compat: Int,
+                      revision: Int,
+                      hasEntryPoints: Boolean)
+
+object NirPrelude {
+  def readFrom(buffer: ByteBuffer): NirPrelude = {
+    val magic    = buffer.getInt()
+    val compat   = buffer.getInt()
+    val revision = buffer.getInt()
+
+    // indicates whether this NIR file has entry points
+    // and thus should be made reachable, no matter
+    // what the reachability algorithm does
+    // example: reflectively instantiatable classes
+    // since: compat = 4, revision = 7
+    val hasEntryPoints = {
+      import scala.math.Ordering.Implicits._
+      if ((compat, revision) < (4, 7))
```

Comment:
Once the assertions on `magic` and `compat` are moved above this point, this test simply becomes
```scala
if (revision < 7)
```